MovieQuiz/Presentation/MovieQuizPresenter.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  MovieQuizPresenter.swift
+//  MovieQuiz
+//
+//  Created by Всеволод Царев on 31.10.2022.
+//
+
+import UIKit
+
+final class MovieQuizPresenter: QuestionFactoryDelegate {
+    private var currentQuestionIndex = 0
+    private let questionsAmount: Int = 10
+    private var correctAnswer = 0
+    private var currentQuestion: QuizQuestion?
+    private weak var viewController: MovieQuizViewControllerProtocol?
+    private var questionFactory: QuestionFactoryProtocol?
+    private let statisticService: StatisticService = StatisticServiceImplementations()
+    
+    init(viewController: MovieQuizViewControllerProtocol) {
+        self.viewController = viewController
+        questionFactory = QuestionFactory(delegate: self, moviesLoader: MoviesLoader())
+        questionFactory?.loadData()
+        viewController.showLoadingIndicator()
+    }
+    
+    func didLoadDataFromServer() {
+        viewController?.hideLoadingIndicator()
+        questionFactory?.requestNextQuestion()
+    }
+    
+    func didFailToLoadData(with error: Error) {
+        let message = error.localizedDescription
+        viewController?.showNetworkError(message: message)
+    }
+    
+    func didReceiveNextQuestion(question: QuizQuestion?) {
+        guard let question = question else {
+            return
+        }
+        currentQuestion = question
+        let viewModel = convert(model: question)
+        DispatchQueue.main.async { [weak self] in
+            self?.viewController?.show(quiz: viewModel)
+        }
+    }
+    
+    func yesButtonAction() {
+        didAnswerAction(isYes: true)
+    }
+    
+    func noButtonAction() {
+        didAnswerAction(isYes: false)
+    }
+    
+    func asyncWaitOneSecond(escaping: @escaping () -> Void) {
```

Comment:
принято такой параметр называть completion
```suggestion
    func asyncWaitOneSecond(completion: @escaping () -> Void) {
```